ooniapi/services/ooniprobe/src/ooniprobe/routers/reports.py

```diff
@@ -0,0 +1,256 @@
+    if len(asn) > 12 or len(asn) < 3 or not asn.startswith("AS"):
+        asn = "AS0"
+    try:
+        asn_i = int(asn[2:])
+    except Exception:
+        asn_i = 0
+
+    cc = data.probe_cc.upper().replace("_", "")
+    if len(cc) != 2:
+        cc = "ZZ"
+    test_name = data.test_name.lower()
+    rid = generate_report_id(test_name, settings, cc, asn_i)
+    resp = OpenReportResponse(
+        backend_version="1.3.5", supported_formats=["yaml", "json"], report_id=rid
+    )
+    setnocacheresponse(response)
+    return resp
+
+
+class ReceiveMeasurementResponse(BaseModel):
+    """
+    Acknowledge
+    """
+
+    measurement_uid: str | None = Field(
+        examples=["20210208220710.181572_MA_ndt_7888edc7748936bf"], default=None
+    )
+
+
+@timer
+@router.post("/report/{report_id}", tags=["reports"])
+async def receive_measurement(
+    report_id: str,
+    request: Request,
+    response: Response,
+    cc_reader: CCReaderDep,
+    asn_reader: ASNReaderDep,
+    settings: SettingsDep,
+    s3_client: S3ClientDep,
+    content_encoding: str = Header(default=None),
+) -> ReceiveMeasurementResponse | Dict[str, Any]:
+    """
+    Submit measurement
+    """
+    setnocacheresponse(response)
+    empty_measurement = {}
+    try:
+        rid_timestamp, test_name, cc, asn, format_cid, rand = report_id.split("_")
+    except Exception:
+        log.info("Unexpected report_id %r", report_id[:200])
+        raise error("Incorrect format")
+
+    # TODO validate the timestamp?
+    good = len(cc) == 2 and test_name.isalnum() and 1 < len(test_name) < 30
+    if not good:
+        log.info("Unexpected report_id %r", report_id[:200])
+        error("Incorrect format")
+
+    try:
+        asn_i = int(asn)
+    except ValueError:
+        log.info("ASN value not parsable %r", asn)
+        error("Incorrect format")
+
+    if asn_i == 0:
+        log.info("Discarding ASN == 0")
+        Metrics.MSMNT_DISCARD_ASN0.inc()
+        return empty_measurement
+
+    if cc.upper() == "ZZ":
+        log.info("Discarding CC == ZZ")
+        Metrics.MSMNT_DISCARD_CC_ZZ.inc()
+        return empty_measurement
+
+    data = await request.body()
+    if content_encoding == "zstd":
+        try:
+            data = zstd.decompress(data)
+            ratio = len(data) / len(data)
+            log.debug(f"Zstd compression ratio {ratio}")
+        except Exception as e:
+            log.info("Failed zstd decompression")
+            error("Incorrect format")
+
+    # Write the whole body of the measurement in a directory based on a 1-hour
+    # time window
+    now = datetime.now(timezone.utc)
+    h = sha512(data).hexdigest()[:16]
+    ts = now.strftime("%Y%m%d%H%M%S.%f")
+
+    # msmt_uid is a unique id based on upload time, cc, testname and hash
+    msmt_uid = f"{ts}_{cc}_{test_name}_{h}"
+    Metrics.MSMNT_RECEIVED_CNT.inc()
+
+    compare_probe_msmt_cc_asn(cc, asn, request, cc_reader, asn_reader)
+    N_RETRIES = 3
+    for t in range(N_RETRIES):
```

Comment:
If we are going to be doing some retries, we should place some kind of jitter in the retries to avoid a thundering herd from the workers. 

I would suggest we do an exponential backoff with jitter using something like this:
```
sleep_amount = random.uniform(0, min(3, 0.3 * 2 ** t))
await asyncio.sleep(sleep_amount)
```

This should give us an exponential backoff with some jitter up to 3 seconds maximum.